pkg/cmd/pulumi/schema_generate_sdk.go

```diff
@@ -0,0 +1,164 @@
+		Args:  cmdutil.ExactArgs(1),
+		Short: "Generate language SDKs from a Pulumi package schema",
+		Long: "Generate language SDKs from a Pulumi package schema.\n" +
+			"\n" +
+			"Given a Pulumi package schema, write out language SDKs for the package.\n" +
+			"The schema is validated as if by `pulumi schema check` prior to generation.\n" +
+			"\n" +
+			"Valid targets are dotnet, go, nodejs, and python. The 'all' target will\n" +
+			"generate an SDK for each suported target.",
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			if outputDirectory == "" {
+				return fmt.Errorf("an output directory must be specified")
+			}
+
+			pkgSpec, err := loadSchemaSpec(args[0], "")
+			if err != nil {
+				return err
+			}
+
+			pkg, diags, err := schema.BindSpec(pkgSpec, nil)
+			if len(diags) != 0 {
+				diagWriter := hcl.NewDiagnosticTextWriter(os.Stderr, nil, 0, true)
+				wrErr := diagWriter.WriteDiagnostics(diags)
+				contract.IgnoreError(wrErr)
+
+				if diags.HasErrors() {
+					return fmt.Errorf("too many errors")
+				}
+			}
+			if err != nil {
+				return err
+			}
+
+			var files map[string][]byte
+			switch target {
+			case "all":
+				files, err = genAllSDKs(pkg)
+			case "dotnet":
+				files, err = genDotnetSDK(pkg)
+			case "go":
+				files, err = genGoSDK(pkg)
+			case "nodejs":
+				files, err = genNodeSDK(pkg)
+			case "python":
+				files, err = genPythonSDK(pkg)
+			default:
+				return fmt.Errorf("unknown target '%v'", target)
+			}
+			if err != nil {
+				return err
+			}
+
+			if err = os.MkdirAll(outputDirectory, 0755); err != nil {
+				return err
+			}
+			paths := make([]string, 0, len(files))
+			for p := range files {
+				paths = append(paths, p)
+			}
+			sort.Strings(paths)
+			for _, path := range paths {
+				realPath := filepath.Join(outputDirectory, filepath.FromSlash(path))
+				if err = os.MkdirAll(filepath.Dir(realPath), 0755); err != nil {
+					return err
+				}
+				if err = os.WriteFile(realPath, files[path], 0600); err != nil {
+					return err
+				}
+			}
+
+			return nil
+		}),
+	}
+
+	cmd.PersistentFlags().StringVarP(&target, "target", "t", "all",
+		"the target language for SDK generation")
+	cmd.PersistentFlags().StringVarP(&outputDirectory, "output-directory", "o", "",
+		"the output directory for the generated SDK")
+
+	return cmd
+}
+
+const toolName = "'pulumi schema generate'"
+
+func genAllSDKs(pkg *schema.Package) (map[string][]byte, error) {
+	files := map[string][]byte{}
+
+	gens := []struct {
+		name string
+		gen  func(pkg *schema.Package) (map[string][]byte, error)
+	}{
+		{name: "dotnet", gen: genDotnetSDK},
+		{name: "go", gen: genGoSDK},
+		{name: "nodejs", gen: genNodeSDK},
+		{name: "python", gen: genPythonSDK},
+	}
+	for _, g := range gens {
+		sdk, err := g.gen(pkg)
+		if err != nil {
+			return nil, fmt.Errorf("failed to generate %v SDK: %w", g.name, err)
+		}
+		for file, contents := range sdk {
+			files[path.Join(g.name, file)] = contents
```

Comment:
We may already be doing this, but this map will get quite large for some of our providers (e.g., azure-native).

We might consider splitting this up by language, or dumping to disk as we go.